packages/nuxt/src/core/nuxt.ts

```diff
@@ -923,12 +923,20 @@ async function resolveModules (nuxt: Nuxt) {
   // Loop layers in reverse order, so that the extends are loaded first and project is the last
   const configs = nuxt.options._layers.map(layer => layer.config).reverse()
   for (const config of configs) {
+    // Check if this is the root project (modules from root project cannot be disabled)
+    const isRootProject = config.rootDir === nuxt.options.rootDir
```

Comment:
I don't think we need to track modules from layers - any module should be able to be disabled